### ISSUE TITLE
[BUG] Multiple Region Parallel Queries

### ISSUE THREAD

**Issue by mgwidmann** [dynamoose/dynamoose#1002]
### Summary:
The suggested fix for being able to switch regions on the fly will succeed in low volume environments, but it does not work as expected when 2 or more queries are ongoing at the same time.

https://stackoverflow.com/questions/61101104/switch-dynamodb-region-on-the-fly

The reason for this it seems, is the Dynamo DB client is shared for all queries. Each model, or each query, does not have the capability to specify its own `new AWS.DynamoDB()` client. Here is where the client is cached:

https://github.com/dynamoose/dynamoose/blob/master/lib/aws/ddb/index.ts#L3

If no client is ever set, a new client is used for each invocation, but once set it will reuse that same client for every query. This makes it impossible to change the region in a parallel execution environment without affecting other queries.

### Code sample:
This particular section does not apply to this issue, and I cannot share this information. To reproduce this issue, you need two queries firing concurrently, either via multiple HTTP requests or within the same HTTP request. We achieved it through using GraphQL, which fires every resolver in parallel, so when two resolvers hit dynamo and one of them is switching the region it will randomly fail with the below exception.

#### Schema
```
// Code here
```

#### Model
```
// Code here
```

#### General
```
// Code here
```

### Current output and behavior (including stack trace):
```
(node:28150) UnhandledPromiseRejectionWarning: ResourceNotFoundException: Requested resource not found
    at Request.extractError (/Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/protocol/json.js:51:27)
    at Request.callListeners (/Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/request.js:688:14)
    at Request.transition (/Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/request.js:690:12)
    at Request.callListeners (/Users/ruhiparv/workplace/Quarry/src/NodeJS-QuarryLambda/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at emitUnhandledRejectionWarning (internal/process/promises.js:168:15)
    at processPromiseRejections (internal/process/promises.js:247:11)
    at processTicksAndRejections (internal/process/task_queues.js:94:32)
```

### Expected output and behavior:
Dynamo does not crash. If the `AWS.DynamoDB` client used for each query could be specified manually, there could be a workaround for this. For example, if the following code worked:

```javascript
let db1 = MyModelDb.withOptions({ ddb: new AWS.DynamoDB({ region: "us-east-1" }) });
let db2 = MyModelDb.withOptions({ ddb: new AWS.DynamoDB({ region: "us-west-2" }) });

await Promise.all([db1.query(...).exec(), db2.query(...).exec()]);
```

### Environment:

Operating System: Mac OSX
Operating System Version: 10.14.6
Node.js version (`node -v`): 14.7.0
NPM version: (`npm -v`): 6.14.7
Dynamoose version: 2.2.1

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose
   - (Version 2.2.1 but I this issue is not fixed in the latest as well)

**Comment by fishcharlie** [comment]
There likely needs to be more definition around what this API and syntax should look like. But for sure a feature request and not a bug.

**Comment by simlu** [comment]
+1 Would be great to have different dynamoose instances that have different configurations

**Comment by fishcharlie** [comment]
Rough idea of how I **think** this going to work:

- `dynamoose.instance` will be a new class. So you will be able to run `const west = new dynamoose.instance()`
- Within that instance, all Dynamoose functions and properties will be accessible. So you can call the same `(new dynamoose.instance()).model("User")` methods and such to create models within that instance.
- Only different things about instances will be the ModelStore, and AWS related methods. Each instance will hold reference to it's own instance of the AWS related stuff and ModelStore.
- When importing/requiring `dynamoose`, it will create a "default" instance for you. So all existing methods will just work (and this will not be a breaking change).

Couple of other things:

- I've been considering adding an option to copy models between instances. But I'm not sure that is something that would be beneficial or useful. If anyone has thoughts about this would love to hear it.
- I'm also not sure how beneficial it is to have to keep track of a different model/instance for every different region. So not sure if there is a better way to handle that. Maybe hearing more about people's use cases would be helpful here.
- Finally, in the original issue it has an example: `await Promise.all([db1.query(...).exec(), db2.query(...).exec()]);`. Would it be useful for people to have a way for Dynamoose to handle these multi region requests for you that is built in? I'd love to hear more about the use cases around this and what exactly you are looking to do with it so we can make sure the API is the best it can be.

I'm aiming to have this work be part of v3 (#1040). Right now I think it's going to be implemented earlier on in the v3 development process. So hopefully there'll be beta and alpha versions with this functionality out soon!!

**Comment by mohemish9** [comment]
@fishcharlie Can you kindly provide examples of how to switch between instances? 
I looked up the documentation, but the only example was how to create the instance. Some instructions on how to use instances would be greatly appreciated. 

The original example by @mgwidmann was:
```
let db1 = MyModelDb.withOptions({ ddb: new AWS.DynamoDB({ region: "us-east-1" }) });
let db2 = MyModelDb.withOptions({ ddb: new AWS.DynamoDB({ region: "us-west-2" }) });

await Promise.all([db1.query(...).exec(), db2.query(...).exec()]);
```
However, I couldn't find withOptions method in the documentation. How would you switch between db1 and db2 in the current implementation?

I really appreciate your help!

**Comment by fishcharlie** [comment]
@mohemish9 Once you create an instance you can use https://dynamoosejs.com/guide/Instance#new-instancetablename-models to create a `Table` instance using that model. Currently there is no way to use it with the [Model initializer](https://dynamoosejs.com/guide/Model#dynamoosemodelitemname-schema-options) (but if that is something you're interested in, please submit a new issue as a feature request).

**Comment by mohemish9** [comment]
@fishcharlie Thank you for your quick reply. 
Is it possible to use a table instance to save an item?
